telegram/ext/filters.py

```diff
@@ -65,6 +68,15 @@ def __or__(self, other):
     def __invert__(self):
         return InvertedFilter(self)
 
+    def __str__(self):
+        # Do not rely on classes overwriting __init__ to set a name
+        # so we can keep backwards compatibility
```

Comment:
Currently it's using`self.__class__.__name__` only if no other name is found. The reason we don't use that is that `telegram.ext.Filters.status_update.migrate.__class__.__name__` is `_Migrate` and not `migrate` as we want. So we have to overwrite it.
Or did I misunderstand your question?
I would prefer if it did use `self.__class__.__name__` when a better name is not found though. So if we had name as a class attribute and defaulted it to None (where it would then do `self.__class__.__name__`) would that work?